agent/agent.go

```diff
@@ -460,11 +540,18 @@ func (a *Agent) gatherLoop(
 
 	for {
 		select {
+		case status := <-a.inputStatus[input.Config.Name]:
```

Comment:
Didn't you guys mention that there was an issue if you returned out of the gather loop while the accumulator was still running? Why don't we need to make sure that the accumulator isn't running anymore?

Also related to my other comment on this channel, it could just be (or something like this)

```
case <- a.pluginShutdowns["inputs"][input.Config.Name]:
   log.Println(...)
   delete(...)
   return(
}

Reply:
We found out that that was probably an unrelated issue. This actually works and will not have any errors with an accumulator trying to write to a channel that no longer exists.